app/src/main/java/com/example/groceriesmanager/Activities/EditRecipeActivity.java

```diff
@@ -64,20 +78,80 @@ public void onClick(View v) {
             @Override
             public void onClick(View v) {
                 // todo: check if required fields are filled then save in server
+                String title = etRecipeTitle.getText().toString().trim();
+                if (Objects.equals(title, "")){
+                    Toast.makeText(EditRecipeActivity.this, "Type in a Recipe title.", Toast.LENGTH_LONG).show();
+                }
+                else{
+                    Recipe newUserRecipe = new Recipe();
+                    newUserRecipe.setUser(ParseUser.getCurrentUser());
+                    newUserRecipe.setType("user");
+                    newUserRecipe.setTitle(title);
+
+                    String link = etLink.getText().toString().trim();
+                    if (!Objects.equals(link, "")){
+                        newUserRecipe.setHyperlink_url(link);
+                    }
+
+                    if (checkboxVegetarian.isChecked()){
+                        filtersList.add("vegetarian");
```

Comment:
Perhaps down the road using a bitfield would be a more memory conscience way of doing this instead of strings and a list.  You could just have an int and set various bits based on an enum value and some bit shifting.  This would allow for quick additions of new filters and a quick bit wise check to see if a filter is set or not instead of doing a string compare.  Something to think about if you're looking to do some optimizations or improvements.